### ISSUE TITLE
Initial sync not working on 1.5.6 and synapse v1.69.0

### ISSUE THREAD

**Issue by v411e** [vector-im/element-android#7516]
### Steps to reproduce

1. Make a fresh install of 1.5.6 from GitHub releases
2. Sign in into a homeserver with synapse v1.69.0 running
3. See the popup "Waiting for server response" for several minutes without any changes

But:
Uninstalled 1.5.6 and reinstalled version 1.5.4 - no problems with new sign in
Element Web - no problems with new sign in
FluffyChat - no problems with new sign in

#### What did you expect?
Working initial sync

#### What happened instead?
Sync was stuck

### Your phone model

Oneplus 8T

### Operating system version

Android 12

### Application version and app store

1.5.6 apk from GitHub releases

### Homeserver

keks.club

### Will you send logs?

No

### Are you willing to provide a PR?

No

**Comment by akoyaxd** [comment]
Updated to 1.5.6 form 1.5.4 and I get stuck on Launchscreen. Tested Synapse 1.68.0 and 1.70.1

**Comment by v411e** [comment]
Yeah, I also first got stuck on the Launchscreen after upgrading

**Comment by jmartinesp** [comment]
Hi, I'm sorry for the inconvenience. Could you provide some logs so we can pinpoint the issue? We'll try to reproduce it on our side too.

**Comment by jmartinesp** [comment]
I tried with synapse 1.71.0rc1 (develop) and I got this:

```
11-04 09:05:11.606  3937  4082 D SYNC/SyncTask: INIT_SYNC Perform server request -- END duration: 58 ms
11-04 09:05:11.606  3937  4082 W SYNC/SyncTask: INIT_SYNC request failure: org.matrix.android.sdk.internal.session.sync.DefaultSyncTask@8b0a053
11-04 09:05:11.607  3937  4082 E SyncAndroidService: ## Sync: sync service did fail true
11-04 09:05:11.607  3937  4082 E SyncAndroidService: ServerError(error=MatrixError(code=M_UNKNOWN, message=Additional properties are not allowed ('unread_thread_notifications' was unexpected)
11-04 09:05:11.607  3937  4082 E SyncAndroidService:
11-04 09:05:11.607  3937  4082 E SyncAndroidService: Failed validating 'additionalProperties' in schema['properties']['room']['properties']['timeline']:
11-04 09:05:11.607  3937  4082 E SyncAndroidService:     {'additionalProperties': False,
11-04 09:05:11.607  3937  4082 E SyncAndroidService:      'properties': {'contains_url': {'type': 'boolean'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'include_redundant_members': {'type': 'boolean'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'lazy_load_members': {'type': 'boolean'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'limit': {'type': 'number'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'not_rooms': {'$ref': '#/definitions/room_id_array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'not_senders': {'$ref': '#/definitions/user_id_array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'not_types': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                   'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'org.matrix.labels': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                           'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'org.matrix.not_labels': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                               'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'related_by_rel_types': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                              'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'related_by_senders': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                            'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'rooms': {'$ref': '#/definitions/room_id_array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'senders': {'$ref': '#/definitions/user_id_array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'types': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                               'type': 'array'}},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:      'type': 'object'}
11-04 09:05:11.607  3937  4082 E SyncAndroidService:
11-04 09:05:11.607  3937  4082 E SyncAndroidService: On instance['room']['timeline']:
11-04 09:05:11.607  3937  4082 E SyncAndroidService:     {'unread_thread_notifications': True}, consentUri=null, limitType=null, adminUri=null, retryAfterMillis=null, isSoftLogout=null, newLookupPepper=null, session=null, completedStages=null, flows=null, params=null), httpCode=400)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: 	at org.matrix.android.sdk.internal.network.RetrofitExtensionsKt.toFailure(RetrofitExtensions.kt:103)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: 	at org.matrix.android.sdk.internal.network.RetrofitExtensionsKt.toFailure(RetrofitExtensions.kt:59)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: 	at org.matrix.android.sdk.internal.session.sync.DefaultSyncTask.downloadInitSyncResponse(SyncTask.kt:224)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: 	at org.matrix.android.sdk.internal.session.sync.DefaultSyncTask.access$downloadInitSyncResponse(SyncTask.kt:64)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: 	at org.matrix.android.sdk.internal.session.sync.DefaultSyncTask$downloadInitSyncResponse$1.invokeSuspend(Unknown Source:15)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: 	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: Caused by: java.lang.RuntimeException: MatrixError(code=M_UNKNOWN, message=Additional properties are not allowed ('unread_thread_notifications' was unexpected)
11-04 09:05:11.607  3937  4082 E SyncAndroidService:
11-04 09:05:11.607  3937  4082 E SyncAndroidService: Failed validating 'additionalProperties' in schema['properties']['room']['properties']['timeline']:
11-04 09:05:11.607  3937  4082 E SyncAndroidService:     {'additionalProperties': False,
11-04 09:05:11.607  3937  4082 E SyncAndroidService:      'properties': {'contains_url': {'type': 'boolean'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'include_redundant_members': {'type': 'boolean'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'lazy_load_members': {'type': 'boolean'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'limit': {'type': 'number'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'not_rooms': {'$ref': '#/definitions/room_id_array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'not_senders': {'$ref': '#/definitions/user_id_array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'not_types': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                   'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'org.matrix.labels': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                           'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'org.matrix.not_labels': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                               'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'related_by_rel_types': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                              'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'related_by_senders': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                            'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'rooms': {'$ref': '#/definitions/room_id_array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'senders': {'$ref': '#/definitions/user_id_array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'types': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                               'type': 'array'}},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:      'type': 'object'}
11-04 09:05:11.607  3937  4082 E SyncAndroidService:
11-04 09:05:11.607  3937  4082 E SyncAndroidService: On instance['room']['timeline']:
11-04 09:05:11.607  3937  4082 E SyncAndroidService:     {'unread_thread_notifications': True}, consentUri=null, limitType=null, adminUri=null, retryAfterMillis=null, isSoftLogout=null, newLookupPepper=null, session=null, completedStages=null, flows=null, params=null)
11-04 09:05:11.607  3937  4082 E SyncAndroidService: 	... 13 more
11-04 09:05:11.611  3937  3937 I SyncAndroidService: ## Sync: onDestroy() [im.vector.app.core.services.VectorSyncAndroidService@704fe8c] periodic:false preventReschedule:false
```

**Comment by GreenReaper** [comment]
Not sure if it is the same issue, but I updated the beta client on Android 10 (Redmi Note Pro 6; LineageOS) and I only get new messages via notifications; the messages are not appearing when I open the app, only those from before the update.

I can send to a channel and it appears on my client, but I get no read receipta. When I open a channel, the update cycling bar persists for ~10 seconds. I have sent a bug report from the client with logs.

**Comment by jmartinesp** [comment]
@GreenReaper according to the contents of the rageshake, it looks like it's the same issue.

**Comment by GreenReaper** [comment]
Thanks for letting me know. FWIW I toggled MSC3061 off in labs, and that seems to have forced a sync that received the missing messages, but this is a low traffic server so I don't know if that resolved the general issue of not receiving new messages.

**Comment by squahtx** [comment]
The error
```
11-04 09:05:11.607  3937  4082 E SyncAndroidService: Failed validating 'additionalProperties' in schema['properties']['room']['properties']['timeline']:
11-04 09:05:11.607  3937  4082 E SyncAndroidService:     {'additionalProperties': False,
11-04 09:05:11.607  3937  4082 E SyncAndroidService:      'properties': {'contains_url': {'type': 'boolean'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'include_redundant_members': {'type': 'boolean'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'lazy_load_members': {'type': 'boolean'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'limit': {'type': 'number'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'not_rooms': {'$ref': '#/definitions/room_id_array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'not_senders': {'$ref': '#/definitions/user_id_array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'not_types': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                   'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'org.matrix.labels': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                           'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'org.matrix.not_labels': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                               'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'related_by_rel_types': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                              'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'related_by_senders': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                                            'type': 'array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'rooms': {'$ref': '#/definitions/room_id_array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'senders': {'$ref': '#/definitions/user_id_array'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                     'types': {'items': {'type': 'string'},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:                               'type': 'array'}},
11-04 09:05:11.607  3937  4082 E SyncAndroidService:      'type': 'object'}
```
is coming from Synapse because `unread_thread_notifications` was specified in the sync filter, but older versions of Synapse do not understand it.

**Comment by squahtx** [comment]
Filed as https://github.com/matrix-org/synapse/issues/14365

**Comment by squahtx** [comment]
On the Synapse side, we ought to be more accepting of unrecognized fields in /sync filters (https://github.com/matrix-org/synapse/issues/14365).

On the Element Android side, we can determine whether it's okay to use the `unread_thread_notifications` filter option by checking for `"v1.4"` in `/_matrix/client/versions`'s list of supported spec versions.

**Comment by SapuSeven** [comment]
Is there a workaround for this issue? I use the App daily and this makes it unusable.

**Comment by jmartinesp** [comment]
> Is there a workaround for this issue? I use the App daily and this makes it unusable.

@GreenReaper mentioned above:
> FWIW I toggled MSC3061 off in labs, and that seems to have forced a sync that received the missing messages
But I haven't been able to verify if this fixes or work arounds the issue.

I think for now the best option is to go back to the previous version (1.5.4 or before). We're working on a fix, but it might take a while to be published.

**Comment by v411e** [comment]
The bug just made it into the Playstore and auto-update broke my Element again ^^

**Comment by bagvand** [comment]
problem exist also with latest Synapse version 1.70.1

**Comment by starcraft66** [comment]
Had to leave the element beta program on google play to downgrade to 1.5.4 and log in on my new phone.

**Comment by kylrth** [comment]
It's a big problem that this bug was caught but not stopped from arriving to the main Play Store channel. The app is completely unusable for a large number of people now.

**Comment by kylrth** [comment]
Is this resolved with a newer version of synapse?

**Comment by starcraft66** [comment]
@kylrth I upgraded my home server in attempt to unblock some of my users and that fixed it for them.

**Comment by bmarty** [comment]
> It's a big problem that this bug was caught but not stopped from arriving to the main Play Store channel. The app is completely unusable for a large number of people now. Isn't the point of the beta channel to catch these sorts of bugs before releasing to everyone?

On the PlayStore, the bug is not in the production channel, but only visible by users who join the beta program (i.e. about 10k users).

We are working on a fix before releasing to production.

**Comment by kylrth** [comment]
@bmarty sorry, you're right. It turns out my users who were seeing the problem were also on the beta channel. My bad.

**Comment by karolyi** [comment]
Can confirm with element-android 1.5.6 from FDroid and synapse v1.69.0. Had to downgrade to 1.5.4 to get it working again.

**Comment by jmartinesp** [comment]
This should be fixed on the app side on `v1.5.7` which is already available on Play Store and GH releases page, it will soon come to F-droid too. Please re-open if the issue appears again.

**Comment by akoyaxd** [comment]
After updating to v1.5.7 from the play store I am again stuck in the launch screen. I can't use rage shake there..

**Comment by SapuSeven** [comment]
For me, the updated version works again for messaging, but i haven't tested the login

**Comment by felschr** [comment]
I, too, get stuck on the launch screen after updating from 1.5.3 to 1.5.7 from Google Play like @akoyaxd.
I'm running CalyxOS 4.2.7 (Android 13) on a Pixel 3.

**Comment by jmartinesp** [comment]
That seems like a different issue, could you open a new issue with that info/logs? Thank you, and sorry for the inconvenience.

**Comment by akoyaxd** [comment]
Yes, I will open one

**Comment by akoyaxd** [comment]
New issue: #7557
